src/expr/src/scalar/func.rs

```diff
@@ -1333,11 +1334,10 @@ pub trait TimestampLike: chrono::Datelike + chrono::Timelike + for<'a> Into<Datu
     }
 
     fn truncate_week(&self) -> Self {
-        let num_days_from_monday = self.date().weekday().num_days_from_monday();
-        Self::new(
-            NaiveDate::from_ymd(self.year(), self.month(), self.day() - num_days_from_monday),
-            NaiveTime::from_hms(0, 0, 0),
-        )
+        let num_days_from_monday = self.date().weekday().num_days_from_monday() as i64;
+        let new_date = NaiveDate::from_ymd(self.year(), self.month(), self.day())
+            .add(Duration::days(-num_days_from_monday));
```

Comment:
We should probably be using `checked_sub_signed` here and returning an overflow error. But feel free to just punt to a follow-on issue; there are all sorts of similar errors lurking and we might as well try to solve them all at once rather than one-by-one.